rllib/evaluation/worker_set.py

```diff
@@ -252,6 +254,56 @@ def reset(self, new_remote_workers: List[ActorHandle]) -> None:
         """
         self._remote_workers = new_remote_workers
 
+    def remove_failed_workers(self):
+        faulty_indices = self._worker_health_check()
+
+        # Terminate faulty workers.
+        for worker_index in faulty_indices:
+            worker = self.remote_workers()[worker_index - 1]
+            logger.info(f"Trying to terminate faulty worker {worker_index}.")
+            try:
+                worker.__ray_terminate__.remote()
+            except Exception:
+                logger.exception("Error terminating faulty worker.")
+
+        if len(faulty_indices) == len(self.remote_workers()):
+            raise RuntimeError("Not enough healthy workers remain to continue.")
+
+        # Remove all faulty workers from self._remote_workers.
+        for worker_index in reversed(faulty_indices):
+            del self._remote_workers[worker_index - 1]
+        # TODO: Should we also change each healthy worker's num_workers counter and
+        #  worker_index property?
```

Comment:
why don't you raise above RuntimeError with a safer check like
```
if not self._remote_workers:
    raise RuntimeError("Not enough healthy workers remain to continue.")
```
?

Reply:
done